test/fake_cloud_firestore_test.dart

```diff
@@ -1354,6 +1354,24 @@ void main() {
         expect(snapshot.docs.first.data()!.title, equals(MovieTitle));
       }));
     });
+
+    test('Replace Null with Map', () async {
+      final firestore = FakeFirebaseFirestore();
+
+      var testMap = {'testVal': null};
+      await firestore
+          .collection('testCollection')
+          .doc('testDoc')
+          .update(testMap);
+
+      var newMap = {
+        'testVal': {'innerKey': 'innerVal'}
+      };
+      await firestore
```

Comment:
A unit test should have some expectation. In your test, you implicitly expect the update not to throw an exception. But it won't be clear to an uninformed reader. To make it explicit, you could write it as:

```dart
expect(firestore....update(newMap), completes);
```

or possibly (but less readable):
```dart
expect(() async => await firestore...update(newMap), returnsNormally);
```

Note: I haven't tested this myself.

See
* https://api.flutter.dev/flutter/package-test_api_expect/completes.html
* https://api.flutter.dev/flutter/package-matcher_matcher/returnsNormally-constant.html.

Reply:
Thanks for the feedback, I wasn't sure what the best practice for an expect statement was, when I'm just expecting the function to complete without an exception, so I appreciate the info. Hopefully this is now fixed